pallets/pools/src/lib.rs

```diff
@@ -628,6 +684,42 @@ pub mod pallet {
 				None => None,
 			};
 
+			for (tranche, tranche_input) in tranches.tranches.iter().zip(&tranche_inputs) {
+				let token_name: BoundedVec<u8, T::MaxTokenNameLength> = tranche_input
+					.clone()
+					.metadata
+					.token_name
+					.try_into()
+					.map_err(|_| Error::<T>::BadMetadata)?;
+
+				let token_symbol: BoundedVec<u8, T::MaxTokenSymbolLength> = tranche_input
+					.clone()
+					.metadata
+					.token_symbol
+					.try_into()
+					.map_err(|_| Error::<T>::BadMetadata)?;
+
+				let decimals = match T::AssetRegistry::metadata(&currency) {
+					Some(metadata) => metadata.decimals,
+					None => return Err(Error::<T>::BadMetadata.into()),
```

Comment:
I think we should make this a different error, since the user action for it is different (fixing the currency metadata vs. submitting correct tranche metadata)

Reply:
Yes exactly ^